packages/styled-dropdown-component/src/components/DropdownMenu.js

```diff
@@ -48,12 +68,75 @@ const DropdownMenu = styled.div`
   background-color: ${(props) => colors(props, 'dropdownMenu').backgroundColor};
   background-clip: padding-box;
   border: ${(props) => border(props, 'dropdownMenu').default} ${(props) => colors(props, 'dropdownMenu').borderColor};
-  ${(props) => borderRadius(props)}
+  ${(props) => borderRadius(props)};
   ${(props) => dropdownMenuHidden(props)};
+  ${(props) => isLeftOrRight(props)};
+  ${(props) => isFullWidth(props)};
 `;
 
+StyledDiv.defaultProps = {
+  theme: DefaultTheme,
+  hidden: false,
+  fullWidth: false,
+};
+
+class DropdownMenu extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.handleDocumentClick = this.handleDocumentClick.bind(this);
+
+    this.container = React.createRef();
+  }
+
+  componentDidMount() {
+    this.addEvents();
+  }
+
+  componentWillUnmount() {
+    this.removeEvents();
+  }
+
+  addEvents() {
+    document.addEventListener('click', this.handleDocumentClick, true);
+  }
+
+  removeEvents() {
+    document.removeEventListener('click', this.handleDocumentClick, true);
+  }
+
+  handleDocumentClick(e) {
+    const container = this.container.current;
+
+    if (this.props.hidden || container === null) {
+      return;
+    }
+
+    const containerParent = container.parentNode;
+
+    if (
+      (containerParent === e.target || !containerParent.contains(e.target))
+      && container !== e.target
+      && !container.contains(e.target)
+    ) {
+      this.props.toggle();
+    }
+  }
+
+  render() {
+    const { children, ...rest } = this.props;
+
+    return (
+      <StyledDiv ref={this.container} {...rest}>
```

Comment:
Is there a specific reason to use a ref here?

Can we use something like `onClick={() => this.props.toggle()}`